Components/webComponents/cartComponents/freightOptionsComponent.js

```diff
@@ -18,58 +23,54 @@ export class FreightOptions extends HTMLElement {
         </div>
         `;
 
-        let cartFreightApi = new CartApi();
-        let cartFreightResponse = await cartFreightApi.fetchCart();
-        console.log(cartFreightResponse.selectedFreightOption);
+        const freightOptions = new FreightOptionsApi();
+        this.freightOptions = await freightOptions.getFreightOptions();
+
+        this.render();
+    }
 
-        let freightOptions = new FreightOptionsApi();
-        let response = await freightOptions.getFreightOptions();
-        let freightOptionsReturn = await response.json();
+    static get observedAttributes() {
+        return [ "selectedfreightoption" ];
+    }
 
-        this.renderFreightOptions(freightOptionsReturn, cartFreightResponse);
+    attributeChangedCallback(name, oldValue, newValue) {
+        if (name === "selectedfreightoption") {
+            this.selectedFreightOption = newValue;
+            this.render();
+        }
     }
 
-    renderFreightOptions(freightOptions, cartFreightResponse) {
+    render() {
         let renderContainer = this.querySelector("#cart-freight-options");
         renderContainer.innerHTML = "";
-        for (var i = 0; i < freightOptions.length; i++) {
-            let freightOption = freightOptions[i];
-            let freightName;
-            let freightInput = document.createElement("input");
+        for (let i = 0; i < this.freightOptions.length; i++) {
+            const freightOption = this.freightOptions[i];
+            const freightName = this.freightNames[freightOption.name];
+            const freightInput = document.createElement("input");
```

Comment:
When do we need to use `let`?
And when should we use `const`?

Reply:
we can use let in general as this makes it possible to edit or "change" a value inside a variable. Const is used when a value should be constant and not able to be changed!